test/test_tensorexpr.py

```diff
@@ -998,6 +999,21 @@ def run_remainder(x, y):
         y = run_remainder(nans, a)
         np.testing.assert_allclose(x.numpy(), y.numpy())
 
+    def test_remainder_types(self):
+        def do_mod(x, y):
+            return x % y
+
+        inputs = [torch.rand(10, dtype=torch.float),
+                  torch.randint(1, 1000, (10,), dtype=torch.int32),
+                  torch.randint(1, 1000, (10,), dtype=torch.int16)
+                  ]
+
+        scripted = torch.jit.script(do_mod)
+        for (a, b) in itertools.product(inputs, repeat=2):
```

Comment:
This is a very nice exhaustive test for `%` operation. Could we have the same for `torch.remainder` and `torch.fmod` called explicitly rather that through `%`? If we add `nan` and `0` to the `inputs` it could also be combined with the previous test.